nix/static.nix

```diff
@@ -0,0 +1,37 @@
+{
+  lib,
+  pkgsStatic,
+  crane,
+  fenix,
+  nrr,
+}: let
+  target = pkgsStatic.stdenv.hostPlatform.config;
+  targetUpper = lib.toUpper (builtins.replaceStrings ["-"] ["_"] target);
+  inherit (pkgsStatic.stdenv) cc;
+
+  toolchain = with fenix;
+    combine [
+      minimal.cargo
+      minimal.rustc
+      targets.${target}.latest.rust-std
+    ];
```

Comment:
Maybe switch to [rust-overlay](https://github.com/oxalica/rust-overlay)? It fixes some issues that aren't fixed in fenix and also has more convenient APIs.